### DISCUSSION TITLE
Twig | Drupal at your Fingertips

### DISCUSSION THREAD

**Discussion by giscus** [selwynpolit/d9book#126] [Q&A]
# Twig | Drupal at your Fingertips

Drupal at your Fingertips: A developers quick reference for Drupal 9 and 10

https://selwynpolit.github.io/d9book/twig

**Comment by stephencapellic** [comment]
Hello Selwyn! Thank you for this incredible resource. 

We have a policy to referenced entitles (child paragraphs, entity reference fields) independently, meaning that if "foo" is referencing "bar" then I will print {{ content.bar }} and then the "bar" will print as the View Mode defined in the field formatter definition for "bar" on "foo". This allow the site builder to easily control how the field is rendered through the interface, the interface is the single-source of truth for how a field is rendered.

In this section [https://selwynpolit.github.io/d9book/twig#entity-reference-destination-content], I see printing fields on "bar" but they aren't subject to any View Mode settings. Do you know how to do what what's happening in section but instead of accessing the field value directly, instead render that field through a defined View Mode?

**Reply by stephencapellic** [reply]
Of course, the process of writing this empowered me the right prompts to get the answer through Google. Here's an example: https://drupal.stackexchange.com/a/207719/98758. And now here's some code based on my foo/bar example.

{% set bar = content.field_bar["#items"].getValue() %} 
{% for item in bar %} 
  {{ drupal_entity('paragraph', item.target_id, 'VIEW_MODE') }} 
{% endfor %}

**Reply by stephencapellic** [reply]
And more accurately, this what I'm really looking for, because I just want to have field-level control when rendering "bar" in the "foo" Twig file, I think it looks likes like this? I need to test this code but this or something close to this should work?

{% set bar = content.field_bar["#items"].getValue() %}
{% for item in bar %}
    {{ drupal_field('field_title', 'node', bar, 'teaser') }}
{% endfor %}

**Reply by selwynpolit** [reply]
@stephencapellic Did you try that code?  It looks pretty sane to me.

**Comment by stephencapellic** [comment]
I did. It's brilliant!

**Reply by selwynpolit** [reply]
Sweet!  Do you want to add something to the page about this?
